lib/typed/serializer.rb

```diff
@@ -38,6 +38,8 @@ def deserialize_from_creation_params(creation_params)
 
         if value.nil? && !field.default.nil?
           Success.new(Validations::ValidatedValue.new(name: field.name, value: field.default))
+        elsif !field.required && (value.nil? || value == "")
```

Comment:
(nitpick) the if before and the elsif after it have the value check first and then the field check so it's a bit faster to read as if we follow that ordering here `elsif (value.nil? || value == "") && !field.required`